tailor_api/src/util.rs

```diff
@@ -0,0 +1,11 @@
+use serde::{Deserialize, Serialize};
+use std::fmt::Debug;
+
+#[cfg(feature = "test-utilities")]
+pub fn default_json_roundtrip<T: Serialize + for<'a> Deserialize<'a> + Default + Debug + Eq>() -> ()
+{
+    let orig = T::default();
+    let serialized = serde_json::to_string_pretty(&orig).unwrap();
+    let deserialized: T = serde_json::from_str(&serialized).unwrap();
+    assert_eq!(deserialized, orig)
+}
```

Comment:
I always write roundtrip tests for anything I serialize/deserialize and expect to be "roundtrippable".

We should be fine with the derived instances, but a roundtrip test can add some extra security in case one of the types (or a struct's field's type) has a manually defined instance.

The `Default` constraint is just because I was too lazy to write a property test.

Reply:
I understand your concerns, but I'm still not convinced that we need this. For me, it is the responsibility of `serde_json` to make sure serialization and deseralization work as intended. And as long as we don't write manual trait impls for `Serialize` or `Deserialize`, I do not expect any problems with `serde_json`.

And even if we keep this test case, I don't think that `tailor_api` is a good place, because this crate is just there to share type definitions and a little bit of code between `tailord` and `tailor_client`. Probably, `tailord` should implement such tests.

Reply:
Removed.